src/backend/c-lisp.py

```diff
@@ -59,18 +71,21 @@ def gen_function(self, func):
             if not len(elem) == 2:
                 raise CodegenError(f"Bad function prototype: {func[1]}")
 
-        self.symbol_types = {}  # Clear the symbol table
+        # Clear the symbol table and scope stack
+        self.symbol_types = {}
+        self.scopes = [random_label(CLISP_PREFIX, ["scope"])]
+
         name, ret_type = func[1][0]
         parm_types = []
+        header = func[1][0:1]
         for parm in func[1][1:]:
-            parm_types.append(parm[1])
-            self.symbol_types[parm[0]] = parm[1]
+            typ = parm[1]
```

Comment:
and this function doesn't have to be modified

Reply:
Done